src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java

```diff
@@ -298,7 +302,7 @@ private void recursivelyCopyProjectDependencies(
             Project currentProject, DependencySet dependencySet, Map<Configuration, String> copiedConfigurationsCache) {
         dependencySet
                 .matching(dependency -> ProjectDependency.class.isAssignableFrom(dependency.getClass()))
-                .configureEach(dependency -> {
+                .all(dependency -> {
```

Comment:
configureEach is lazy and all is not lazy, and apparently this change wasn't super essential but nice to be defensive anyway.